internal/services/compute/managed_disk_resource.go

```diff
@@ -812,6 +827,7 @@ func resourceManagedDiskRead(d *pluginsdk.ResourceData, meta interface{}) error
 		d.Set("os_type", props.OsType)
 		d.Set("tier", props.Tier)
 		d.Set("max_shares", props.MaxShares)
+		d.Set("hyper_v_generation_version", props.HyperVGeneration)
```

Comment:
is there a default value for this for older disk images? iirc there wasn't - so this'll be null & should be defaulted in the read function is the user doesn't specify a value here

Reply:
The default value when not specified is an empty string (not nil) as the type of props.HyperVGeneration is not a pointer.